packages/eslint-plugin/lib/__tests__/no-codemod-comment.test.ts

```diff
@@ -0,0 +1,71 @@
+import { RuleTester } from 'eslint';
+
+import rule from '../rules/no-codemod-comment';
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 'latest',
+    ecmaFeatures: {
+      jsx: true,
+    },
+  },
+});
+
+ruleTester.run('no-codemod-comment', rule, {
+  valid: [
+    {
+      code: [
+        // This has an invalid header so is ignored
+        `/* integrity: codemod-hash-1399692252 */`,
+        `// TODO codemod-generated-comment signed: codemod-hash-1399692252`,
```

Comment:
Could you explain how the integrity check works?

So if I was to imagine a codemod run on a repo and the diffs raised in a PR, would these be presented as an error? Or can they be merged and subsequent changes to the file would trigger an error?

Reply:
@danieldelcore I should add a README for these rules sorry mate, will amend.

In essence the idea is that a codemod would add a header comment and generate hashes off the comments it introduces to code. Each of these comments are validated against the header. So it's clear that this were generated by a script.

I thought of multiple different approaches here tho for example:

* Using a hash of the whole file from the time that the codemod ran and storing it in a header, then recalculating the hash off the whole file minus the hash. Idea here would be as soon as you make a change in this file it will error. IE don't leave the codemod header / make a change without validating that you've acknowledged the codemod ran.
* Alternatively just erroring or warning off comments generated by codemod.

Or some combination of the above.